juju/juju
# Remove InstanceMutater feature flag and old update charm lxd profile code path

## Description of change

Remove InstanceMutater feature flag and fix a few tests failing due to
change in machine and model manifolds, and uniter worker.

Remove old code to upgrade a charm lxd profile which has been replaced
with the instance mutater worker. Including the instanceCharmProfileData
collection.

## QA steps

Deploy and upgrade charms with and without lxd profiles.  Try bundles
with more than 5 machines, subordinates, containers: with and without 
the lxd provider, local and charm store charms.

Bootstrap 2.5.4.  Deploy a few charms with and without lxd profiles.  
Upgrade the juju controller but not model to this code.  Upgrade charms,
should failure with error message to upgrade the model on charms 
with lxd profiles.  Upgrade the model and try upgrade charms again
without error.

charms with an lxd profile:
./testcharms/charm-repo/quantal/lxd-profile-alt
./testcharms/charm-repo/quantal/lxd-profile-subordinate

juju deploy cs:nova-compute
juju deploy cs:neutron-openvswitch-252
juju add-relation nova-compute:neutron-plugin neutron-openvswitch:neutron-plugin
let settle, will be waiting for other relations
juju upgrade-charm neutron-openvswitch  <-- adds an lxd profile
